### DISCUSSION TITLE
the trait GenerateImplicitData is not implemented for

### DISCUSSION THREAD

**Discussion by webern** [shepmaster/snafu#326] [General]
I am seeing build failures `the trait GenerateImplicitData is not implemented for &str` when all of my immediate dependencies have been updated to 0.7 but one of my transient dependencies is still using 0.6.10. Is this expected? Can only one major version of Snafu exist in a dependency tree?

**Comment by webern** [comment]
Actually having multiple versions in the tree doesn't seem to be the source of the issue since I still have this problem after ensuring that only 0.7 existed in the tree

**Comment by webern** [comment]
I figured this out. One of the variants had a struct field named `location` which conflicted with something that snafu is doing under the hood. The compiler message is not helpful here, but when I figured it out and changed the name of that field the problem ceased.

**Comment by shepmaster** [comment]
Amusingly, I had the same problem when upgrading one of my own projects 😉. It was probably a bit of a stretch to make `location` an automatic field like `source` or `backtrace`; I may consider stepping that decision back on the next semver-incompatible release.

> The compiler message is not helpful here

If I added some text to [`GenerateImplicitData`](https://docs.rs/snafu/latest/snafu/trait.GenerateImplicitData.html) that linked to [`snafu(implicit)`](https://docs.rs/snafu/latest/snafu/derive.Snafu.html#controlling-implicitly-generated-data), would that have helped your case; did you look at that trait's docs?

**Reply by webern** [reply]
I eventually did look at the trait definition and saw something that tipped me off.

Maybe a better solution would be to raise an error when you are parsing the tokens of the user-provided struct during the derive macro execution. If the presence of a location field will always be a problem, you could head it off there by raising an error or panic in the derive macro execution.
